### ISSUE TITLE
Enforce normal case for settings and name members

### ISSUE THREAD

**Issue by gisjedi** [ngageoint/seed#136]
We've run in to multiple consumers that are confused by inconsistent case of their `settings` or input `name` members compared to the injected environment variables. All our environment variables are normalized to only uppercase alphabetic and underscore characters. Normalization is well and good, but it is very confusing to users that what they see in their container at run-time differs from their manifest keys. We should change the next version of the spec to enforce the normal form in the keys.  i.e., All `settings` and input `name` members must only be alphabetic and underscore characters.

**Comment by gisjedi** [comment]
TLDR:

* Remove injection normalization language from specification (3.1.1.2) for all input variables that are defined in the interface.
* Update Files, JSON, Settings and Mount object `name` pattern to `^[a-zA-Z0-9_]+$`

In further discussion on the confusion around this inconsistency between the interface `name` member values and the injected environment variables, we've settled on what should be the best case. We are leaning toward allowing the complete range of values that Linux shells allow, but eliminating the use of dashes. This would drop environment variable normalization in favor of a tighter pattern on all input and setting `name` member values. The pattern proposed is `^[a-zA-Z0-9_]+$`.
